worker/task/process_event_delivery.go

```diff
@@ -44,9 +44,12 @@ func ProcessEventDelivery(appRepo datastore.ApplicationRepository, eventDelivery
 		// Load message from DB and switch state to prevent concurrent processing.
 		m, err := eventDeliveryRepo.FindEventDeliveryByID(context.Background(), Id)
 
+		var delayDuration time.Duration
```

Comment:
Don't you think we can define and assign this once instead of reassigning it multiple times?

Reply:
the mock tests where failing when I did it that way but let me dig deeper and see.

Reply:
Hmm, I think is cos you need to EXPECT the call to the repo function

Reply:
yeah exactly, fixed it it now thanks.